spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8ConfigMapPropertySource.java

```diff
@@ -41,25 +44,21 @@
 	private static final Log LOG = LogFactory.getLog(Fabric8ConfigMapPropertySource.class);
 
 	public Fabric8ConfigMapPropertySource(KubernetesClient client, String name) {
-		this(client, name, null, (Environment) null);
+		this(client, name, null, null);
 	}
 
-	public Fabric8ConfigMapPropertySource(KubernetesClient client, String name, String namespace,
+	public Fabric8ConfigMapPropertySource(KubernetesClient client, String applicationName, String namespace,
 			Environment environment) {
-		super(getName(name, getNamespace(client, namespace)),
-				asObjectMap(getData(client, name, getNamespace(client, namespace), environment)));
-	}
-
-	private static String getNamespace(KubernetesClient client, String namespace) {
-		return StringUtils.isEmpty(namespace) ? client.getNamespace() : namespace;
+		super(getName(applicationName, getNamespace(client, namespace)),
+				getData(client, applicationName, getNamespace(client, namespace), environment));
 	}
 
-	private static Map<String, Object> getData(KubernetesClient client, String name, String namespace,
+	private static Map<String, Object> getData(KubernetesClient client, String applicationName, String namespace,
 			Environment environment) {
 		try {
-			Map<String, Object> result = new LinkedHashMap<>();
-			ConfigMap map = StringUtils.isEmpty(namespace) ? client.configMaps().withName(name).get()
-					: client.configMaps().inNamespace(namespace).withName(name).get();
+			Map<String, Object> result = new HashMap<>();
```

Comment:
I see no reason to have an order for the resulting `Map`, so replace it with `HashMap`. There is no significant change on this, but it creates a mental pressure when reading the code : "if it is a `LinkedHashMap`, means we want to preserve the order", so you are kind of stuck understanding why that order needs to be preserved; when the reality is simply not so.